javascript/src/seedsource/containers/Upload.js

```diff
@@ -0,0 +1,152 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import PropTypes from 'prop-types'
+import shp from 'shpjs'
+import { updateConstraintValues } from '../../actions/constraints'
+import { setError } from '../../actions/error'
+
+
+class Upload extends React.Component {
+    constructor(props) {
+        super(props)
+        this.handleFileUpload = this.handleFileUpload.bind(this)
+        this.state = {
+            isLoading: false
+        }
+    }
+
+
+    handleFileUpload(event) {
+        this.setState({isLoading: true})
+        let files
+        if (event.dataTransfer && event.dataTransfer.files.length) {
+            files = event.dataTransfer.files
+        } else if (event.target) {
+            files = event.target.files
+        }
+
+        let zipFile
+        let shpFile
+        let prjFile
+        for (let i = 0; i < files.length; i += 1) {
```

Comment:
`Array.find` might be useful here:

```javascript
let zipFile = files.find(file => file.name.match(/\.zip/i))
let shpFile = files.find(file => file.name.match(/\.shp$/i))
let prjFile = files.fine(file => file.name.match(/\.prj$/i))
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find